test/test_distributions.py

```diff
@@ -1542,13 +1572,29 @@ def setUp(self):
 
     def test_kl_monte_carlo(self):
         set_rng_seed(0)  # see Note [Randomized statistical tests]
-        for p, q in self.finite_examples:
-            x = p.sample(sample_shape=(23000,))
+        for (p, _), (_, q) in self.finite_examples:
+            x = p.sample(sample_shape=(1000,))
             expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
             actual = kl_divergence(p, q)
-            message = 'Incorrect KL({}, {}). expected {}, actual {}'.format(
-                type(p).__name__, type(q).__name__, expected, actual)
-            self.assertEqual(expected, actual, prec=0.1, message=message)
+            message = 'Incorrect KL({}, {}).\nExpected (Monte Carlo): {}\nActual (analytic): {}'.format(
+                    type(p).__name__, type(q).__name__, expected, actual)
+            try:
+                self.assertEqual(expected, actual, prec=0.1, message=message)
+            except:
+                for sample_extra in self.sampling_extra:
+                    x = torch.cat([x, p.sample(sample_shape=(sample_extra,))], 0)
+                    expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
+                    actual = kl_divergence(p, q)
+                    try:
+                        self.assertEqual(expected, actual, prec=0.1, message=message)
```

Comment:
I don't think it will work to `try...except` here, since the test can never fail. Instead simply
```py
actual = kl_divergence(p, q)
for sample_extra in ...:
    x = torch.cat([x, p.sample(sample_shape=(sample_extra,))], 0)
    expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
    if torch.abs(acutal - expected).max() < precision:
        break
self.assertEqual(ecpected, actual, precision, message)
```